### DISCUSSION TITLE
Difference between #all and #to_a

### DISCUSSION THREAD

**Discussion by burlesona** [jeremyevans/sequel#1803] [General]
I couldn't find any documentation around using `#to_a` on datasets, but in practice it appeared that `#to_a` and `#all` were the same. However, today I ran into a case where, after using `#eager_graph`, `#all` returns model instances, but `#to_a` returns hashes.

Is there any documentation on using `#to_a` versus `#all` and what the expected behavior is for each?

I actually couldn't find any documentation of `#to_a`, so I am assuming that this is actually just the ruby core method being used on an enumerable and thus the behavior may be inconsistent - thus we should always prefer using the `#all` method. Is that correct?

**Comment by jeremyevans** [comment]
`to_a` comes from `Enumerable`, which `Dataset` includes. The eager loading documentation mentions that you must use `all` to get eager loading to work.  `all` is basically `to_a; post_load; each(&block) if block`.  `post_load` here is a private method that handles the eager loading.  In general, it's probably best to use `Dataset#all` and not `Dataset#to_a`
